menuxconfig.py

```diff
@@ -0,0 +1,765 @@
+
+Copyright (c) 2015-2016, Cory Xie cory.xie@gmail.com
+
+Permission to use, copy, modify, and/or distribute this software for
+any purpose with or without fee is hereby granted, provided that
+the above copyright notice and this permission notice appear in all
+copies.
+
+THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
+AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
+DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA
+OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER
+TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+PERFORMANCE OF THIS SOFTWARE.
+"""
+
+try:
+    # for Python2
+    import Tkinter as tk
+    import ttk as ttk
+    import ScrolledText as tkst
+    import tkMessageBox as tkmsgbx
+    import tkFileDialog as tkfile
+except ImportError:
+   # for Python3
+    import tkinter as tk
+    import tkinter.ttk as ttk
+    import tkinter.scrolledtext as tkst
+    import tkinter.messagebox as tkmsgbx
+    import tkinter.filedialog as tkfile
+
+from kconfiglib import Kconfig, \
+    Symbol, Choice, MENU, COMMENT, \
+    BOOL, TRISTATE, STRING, INT, HEX, UNKNOWN, \
+    expr_value, \
+    TRI_TO_STR, standard_config_filename
+
+
+def usage(parser, err_str):
+    """ Helper for printing usage
+    """
+    print(err_str + '\n')
+    parser.print_help()
+
+
+def main():
+    """ Main entry point
+    """
+
+    parser = argparse.ArgumentParser(prog=__file__,
+                                     description=information,
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    optional = parser._action_groups.pop()
+    optional.add_argument(
+        '-k', '--kfile', help="Root Kconfig File", action="store",
+        default='Kconfig')
+    optional.add_argument(
+        '-o', '--ofile', help="Config File to use", action="store",
+        default=standard_config_filename())
+    optional.add_argument(
+        '-t', '--title', help="Title to use", action="store",
+        default='Configuration Utility')
+    optional.add_argument(
+        '-H', '--header', help="Header to use in config file", action="store",
+        default=None)
+    optional.add_argument(
+        '-C', '--tree_closed', help="Keep the tree closed by default", action="store_true",
+        default=False)
+
+    parser._action_groups.append(optional)
+    cmd_args, unknown = parser.parse_known_args()
+
+    kfile = cmd_args.kfile
+
+    if len(unknown) == 1:
+        kfile = ''.join(unknown)
+
+    if len(unknown) > 1:
+        usage("Error: Unknown set of optional arguments: " + str(unknown))
+        sys.exit(1)
+
+    cfile = cmd_args.ofile
+    title = cmd_args.title
+    header = cmd_args.header
+    if cmd_args.tree_closed is True:
+        default_open = False
+    else:
+        default_open = True
```

Comment:
Could do

```python
app = MainxMenuConfigApp(root, conf, cfile, header, not cmd_args.tree_closed)
```

Maybe it could be `tree_open`, to avoid having two variables that are inverses of each other.